src/Schema/Directives/CanDirective.php

```diff
@@ -176,7 +178,7 @@ protected function authorize(Gate $gate, $ability, $model, array $arguments): vo
         // should be [modelClassName, additionalArg, additionalArg...]
         array_unshift($arguments, $model);
 
-        if (! $gate->check($ability, $arguments)) {
+        if (! $gate->forUser(Auth::guard(Authentication::getGuard())->user())->check($ability, $arguments)) {
```

Comment:
This line is too long and does too much.

Reply:
Ok, no problem.